src/app/jscaip/Coord.ts

```diff
@@ -149,28 +149,30 @@ export class Coord extends Vector {
         return new Vector(this.x, this.y);
     }
 
-    public getCoordsToward(c: Coord): Coord[] {
+    public getCoordsTowards(c: Coord, includeStart: boolean = false, includeEnd: boolean = false): Coord[] {
         Utils.assert(c.isAlignedWith(this), 'Should only call getCoordsTowards on aligned coords');
-        if (c.equals(this)) {
-            return [];
+        const coords: Coord[] = [];
+        if (includeStart) {
+            coords.push(this);
+        }
+        if (this.equals(c)) {
+            return coords;
         }
         const dir: Ordinal = this.getDirectionToward(c).get();
         let coord: Coord = this.getNext(dir, 1);
-        const coords: Coord[] = [];
         while (coord.equals(c) === false) {
             coords.push(coord);
             coord = coord.getNext(dir, 1);
         }
+        if (includeEnd) {
+            coords.push(coord);
+        }
         return coords;
     }
 
+    // [this, end]
     public getAllCoordsToward(end: Coord): Coord[] {
```

Comment:
toward/towards, voir ce qui est mieux. SORRY je t'ai poussé au towards peut-être, mais : « Toward is the preferred spelling in the United States and Canada. In other English-speaking countries, such as the United Kingdom and Australia, towards is the more common spelling » (same pour forward/backward je crois)

Reply:
refactoré à moultes endroits